File: amvector/service/schema/system.go
// Package schema
// Date: 2024/6/24 09:37
// Author: Amu
// Description:
package schema

type RebootArgs struct{}

type RebootReply struct{}

type ShutdownArgs struct{}

type ShutdownReply struct{}

type GetDNSArgs struct{}

type GetDNSReply struct {
	DNS []string
}

type SetDNSArgs struct {
	DNS []string
}

type SetDNSReply struct{}

type GetSystemTimeArgs struct{}

type GetSystemTimeReply struct {
	SystemTime string
}

type SetSystemTimeArgs struct {
	SystemTime string
}

type SetSystemTimeReply struct{}

type GetSystemTimeZoneListArgs struct {
}

type GetSystemTimeZoneListReply struct {
	SystemTimeZoneList []string
}

type GetSystemTimeZoneArgs struct{}

type GetSystemTimeZoneReply struct {
	SystemTimeZone string
}

type SetSystemTimeZoneArgs struct {
	SystemTimeZone string
}

type SetSystemTimeZoneReply struct{}

type GetDockerRegistryMirrorsArgs struct{}

type GetDockerRegistryMirrorsReply struct {
	Mirrors []string
}

type SetDockerRegistryMirrorsArgs struct {
	Mirrors []string
}

type SetDockerRegistryMirrorsReply struct{}

File: web/src/interface/system.ts
/**
 * @Author     : Amu
 * @Date       : 2024/06/26 00:04:34
 * @Description:
 */

export interface GetDNSResult {
    dns: string[]
}

export interface SetDNSArgs {
    dns: string[]
}

export interface GetSystemTimeResult {
    system_time: string
}

export interface SetSystemTimeArgs {
    system_time: string
}

export interface GetSystemTimezoneResult {
    system_timezone: string
}

export interface SetSystemTimezoneArgs {
    system_timezone: string
}
